lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1005,9 +1005,9 @@ def configure_cdrom(self, vm_obj):
             if self.params["cdrom"]["type"] == "iso" and ("iso_path" not in self.params["cdrom"] or not self.params["cdrom"]["iso_path"]):
                 self.module.fail_json(msg="cdrom.iso_path is mandatory in case cdrom.type is iso")
 
-            if vm_obj and vm_obj.config.template:
-                # Changing CD-ROM settings on a template is not supported
-                return
+            # if vm_obj and vm_obj.config.template:
```

Comment:
@pdellaert I am not sure about this check. Could you please let me know if it is correct ? Thanks.

Reply:
@pdellaert Could you please check this ?

Reply:
I've looked up if PowerCLI supports it. It is only documented for Get, not for New.
But Set and Remove should work, as they take a CDDrive-Object and not the VM/Template as argument.

So if the python api is the same as the PowerCLI one, it could be assumed, that customizing the cdrom of a template works as long as the template already has a cdrom drive. If it does not, it will fail, as the functionality is only available for VirtualMachine-Objects, but not for Template-Objects.
https://code.vmware.com/doc/preview?id=6330#/doc/CDDrive.html

Reply:
This code only blocks changing configuration of a cdrom on a template. 
What worked with the initial code, and still works with the new code:
- Create a VM from a template and add a cdrom if template doesn't have cdrom
- Create a VM from a template and change cdrom config if template has cdrom
- Create a template from a template and change cdrom config if base template has cdrom

What didn't work with original code: 
- Create a template from a template and add a cdrom if base template doesn't have cdrom: `"msg": "Failed to create a virtual machine : Device requires a controller."`
- Update cdrom config of an existing template - (returns unchanged)

With new code, what i've tested:
- Update cdrom config of an existing template with cdrom: ERROR: `"msg": "The operation is not supported on the object."` (this is also visible in vCenter)
- Add cdrom config to an existing template without cdrom: ERROR: `"msg": "The operation is not supported on the object."` (this is also visible in vCenter)

Executed this in vSphere 6.7 btw

**Conclusion: This change needs to be reverted** - Updating a CDROM (or adding one) on an existing template